types.go

```diff
@@ -1490,9 +1495,22 @@ func makeGoLangTypeLength(ti typeInfo) (int64, bool) {
 		return 0, false
 	case typeBigBinary:
 		return int64(ti.Size), true
+	case typeUdt:
+		switch ti.UdtInfo.TypeName {
+		case "hierarchyid":
+			// https://learn.microsoft.com/en-us/sql/t-sql/data-types/hierarchyid-data-type-method-reference?view=sql-server-ver16
+			return 892, true
+		case "geography":
+		case "geometry":
+			return 2147483647, true
+		default:
+			panic(fmt.Sprintf("not implemented makeGoLangTypeLength for user defined type %s", ti.UdtInfo.TypeName))
```

Comment:
should we open an issue to implement app-supplied callbacks to decode unknown types?

Reply:
That could be a possible way to support other data types, but i think that these are the only ones that were missing from the driver, aside from really new types like the [json](https://learn.microsoft.com/en-us/sql/t-sql/data-types/json-data-type?view=azuresqldb-current&viewFallbackFrom=sql-server-ver16) type